src/renovate.ts

```diff
@@ -77,29 +77,15 @@ class Renovate {
   }
 
   /**
-   * Fetch the host docker group of the GitHub Action runner.
+   * Read the group ID off of the Docker socket file that is being
+   * mounted into the container.
    *
-   * The Renovate container needs access to this group in order to have the
+   * The Renovate container's user needs access to belong to this group in order to have the
    * required permissions on the Docker socket.
    */
   private async getDockerGroupId(): Promise<string> {
-    const groupFile = '/etc/group';
-    const groups = await fs.readFile(groupFile, {
-      encoding: 'utf-8',
-    });
-
-    /**
-     * The group file has `groupname:group-password:GID:username-list` as
-     * structure and we're interested in the `GID` (the group ID).
-     *
-     * Source: https://www.thegeekdiary.com/etcgroup-file-explained/
-     */
-    const match = Renovate.dockerGroupRegex.exec(groups);
-    if (match?.groups?.groupId === undefined) {
-      throw new Error(`Could not find group docker in ${groupFile}`);
-    }
-
-    return match.groups.groupId;
+    const dockerSocketStat = await fs.stat(this.input.dockerSocketHostPath());
+    return dockerSocketStat.gid.toString();
```

Comment:
separate PR

Reply:
If this gets merged, I'll create a separate PR that leverages the stat done as part of verification to also get the gid.